### PR TITLE
WT-9276 Remove support for mixed-mode

### PR DESCRIPTION
**Pull request by GWlodarek**
[Patch build](https://spruce.mongodb.com/version/63576ccee3c33128c45ee095/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

### CODE REVIEW

File: dist/api_data.py [hunk 1]
```diff
@@ -73,9 +73,9 @@ def __ge__(self, other):
         are the default, which ensures that once timestamps are used for a key, they are always
         used, and also that multiple updates to a key never use decreasing timestamps and
         \c never which enforces that timestamps are never used for a table. (The \c always,
-        \c key_consistent, \c mixed_mode and \c ordered choices should not be used, and are
-        retained for backward compatibility.)''',
-        choices=['always', 'key_consistent', 'mixed_mode', 'never', 'none', 'ordered']),
+        \c key_consistent, and \c ordered choices should not be used, and are retained for
+        backward compatibility.)''',
+        choices=['always', 'key_consistent', 'never', 'none', 'ordered']),
```
Comment (kommiharibabu) [inline dist/api_data.py:78]: Removing the option can cause backward compatibility problems when that option is used in older releases and later they try to import the table with that config. Refer to WT-8777 for a similar issue for `key_consistent` being removed and added back.

**Review by kommiharibabu — COMMENTED** [review]
The config option cannot be removed as it can cause backward compatibility problems. Otherwise, the code changes are good.

File: dist/api_data.py (hunk 1 above)
Comment (GWlodarek) [reply dist/api_data.py]: Good to know! I put this back.

**Comment by GWlodarek** [comment]
[Patch build](https://spruce.mongodb.com/version/6359ea520305b919fa21c695/tasks)

**Review by kommiharibabu — APPROVED** [review]
LGTM. Thanks.

**Comment by GWlodarek** [comment]
evergreen merge

**Pull request merged**